Gems/Atom/RPI/Code/Source/RPI.Reflect/Image/AttachmentImageAsset.cpp

```diff
@@ -33,7 +36,29 @@ namespace AZ
 
         const RHI::ClearValue* AttachmentImageAsset::GetOptimizedClearValue() const
         {
-            return m_isClearValueValid ? &m_optimizedClearValue : nullptr;
+            return m_optimizedClearValue.get();
+        }
+
+        const AZ::Name& AttachmentImageAsset::GetName() const
+        {
+            return m_name;
+        }
+
+        RHI::AttachmentId AttachmentImageAsset::GeAttachmentId() const
+        {
+            if (m_isUniqueName)
+            {
+                if (!m_name.IsEmpty())
+                {
+                    return m_name;
+                }
+            }
+            return Name(m_assetId.ToString<AZStd::string>());
+        }
+
+         bool AttachmentImageAsset::HasUniqueName() const
+        {
+            return m_isUniqueName;
```

Comment:
I see above that we check for empty name as well, wouldn't an empty name be considered as non unique? In which case we could use that condition above. Which also ask the question if the creator shouldn't handle that special case.